### PR TITLE
7732 new design for demo page

### PR DESCRIPTION
**Pull request by denpira**
https://codio.myjetbrains.com/youtrack/issue/codio-7732

### CODE REVIEW

File: source/javascripts/partials/_demo.es6 [hunk 1]
```diff
@@ -1,7 +1,61 @@
...
+          + '<div class="user-name">' + videoData[0].user_name + '</div>'
+          + '</div></div></li>');
+        $videoDescriptions[index] = (videoData[0].description);
+        $videoTitles[index] = (videoData[0].title);
+      });
+  });
+};
+
+$(document).ready(() => {
+  $($videoList).on("click", "li", (e) => {
```
Comment (AndreyNikitin) [inline source/javascripts/partials/_demo.es6]: $videoList is already a jquery object, no need to use $()

File: source/javascripts/partials/_demo.es6 [hunk 2]
```diff
@@ -1,7 +1,61 @@
...
+    $('.active-item').removeClass('active-item');
+    $(e.currentTarget).find("div").addClass("active-item");
```
Comment (AndreyNikitin) [inline source/javascripts/partials/_demo.es6]: calling $() forces jquery to do same code over and over again, better initialize variable for such
```
var target = $(e.currentTarget); var player = $('#vm-player');
```
and so on

File: source/stylesheets/pages/_demo.scss [hunk 3]
```diff
@@ -3,283 +3,286 @@
...
-    &.second {
-      padding: 2rem 0 1rem;
-    }
   }
 
-  .blue {
-    background-color: #4574e0;
-    color: white;
-    overflow: hidden;
+  h1 {
```
Comment (AndreyNikitin) [inline source/stylesheets/pages/_demo.scss]: better avoid setting css for html object, try using .class-names for such elements, and set css for them.

Comment (AndreyNikitin) [inline source/stylesheets/pages/_demo.scss]: but ok for now

Comment (denpira) [inline source/stylesheets/pages/_demo.scss]: thanks

File: source/javascripts/partials/_demo.es6 (hunk 1 above)
Comment (denpira) [inline source/javascripts/partials/_demo.es6]: Done

File: source/javascripts/partials/_demo.es6 (hunk 2 above)
Comment (denpira) [inline source/javascripts/partials/_demo.es6]: Done

File: source/javascripts/partials/_demo.es6 [hunk 4]
```diff
@@ -1,31 +1,64 @@
...
+    target.find("div").addClass("active-item");
+    var index = target.data('index');
+    $player.attr('src', 'https://player.vimeo.com/video/' + videoIdlist[index] + '?byline=0&portrait=0');
+    var iframe = $player[0];
+    $player.load(function(e) { iframe.api('play'); });
+    $title.text($videoTitles[index]);
+    $description.text($videoDescriptions[index]);
+  });
+
+  setTimeout(function() {
```
Comment (AndreyNikitin) [inline source/javascripts/partials/_demo.es6]: you use two styles ```function() {}``` and ```() => {}```, choose one.

File: source/javascripts/partials/_demo.es6 [hunk 5]
```diff
@@ -1,31 +1,64 @@
...
+  '165269077',
+  '160137702',
+  '177715465',
+  '177714682'
+]
+
+var $videoDescriptions = new Array(videoIdlist.length);
+var $videoTitles = new Array(videoIdlist.length);
+
+const $videoList = $('.video-list ul');
```
Comment (AndreyNikitin) [inline source/javascripts/partials/_demo.es6]: move this into document.ready block, as this html elements could be hidden during file load.

File: source/javascripts/partials/_demo.es6 (hunk 4 above)
Comment (denpira) [inline source/javascripts/partials/_demo.es6]: Done

File: source/javascripts/partials/_demo.es6 (hunk 5 above)
Comment (denpira) [inline source/javascripts/partials/_demo.es6]: Done

File: source/javascripts/partials/_demo.es6 [hunk 6]
```diff
@@ -1,31 +1,61 @@
...
+          + '<div class="user-name">' + videoData[0].user_name + '</div>'
+          + '</div></div></li>');
+        videoDescriptions[index] = (videoData[0].description);
+        videoTitles[index] = (videoData[0].title);
+      });
+  });
+};
+
+$(document).ready(() => {
+  videoList = $('.video-list ul');
```
Comment (AndreyNikitin) [inline source/javascripts/partials/_demo.es6]: var

Comment (denpira) [inline source/javascripts/partials/_demo.es6]: Done

**Comment by AndreyNikitin** [comment]
:+1:

**Pull request merged**